File: Chunk.c
#include "Chunk.h"

static int GetChunkBufferSize(int numComponents, int numSharedComponents)
{
    int bufferSize = kChunkSize -
                     (sizeof(Chunk) - 4 + numSharedComponents * sizeof(int) + numComponents * sizeof(uint));
    return bufferSize;
}

static int GetSharedComponentOffset(int numSharedComponents)
{
    return kChunkSize - numSharedComponents * sizeof(int);
}

static int GetChangedComponentOffset(int numComponents, int numSharedComponents)
{
    return GetSharedComponentOffset(numSharedComponents) - numComponents * sizeof(uint);
}

File: Chunk.lua
local Chunk = {}
ECS.Chunk = Chunk

ECS.Chunk.kChunkSize = 16 * 1024

function Chunk.GetChunkBufferSize( numComponents, numSharedComponents )
	-- local bufferSize = ECS.Chunk.kChunkSize - (numSharedComponents * sizeof(int) + numComponents * sizeof(uint))
	local bufferSize = ECS.Chunk.kChunkSize - (numSharedComponents * 4 + numComponents * 4)
    return bufferSize
end

return Chunk